### ISSUE TITLE
[DependencyInjection] Circular reference detected for autowired decorator

### ISSUE THREAD

**Issue by BoShurik** [symfony/symfony#28283]
**Symfony version(s) affected**: 4.1.3

**Description**  
With this config
```yaml
App\DI\Foo\FooInterface: '@App\DI\Foo\Foo'

App\DI\Foo\FooDecorator:
    decorates: App\DI\Foo\FooInterface

App\DI\Bar\BarInterface: '@App\DI\Bar\Bar'

App\DI\Bar\BarDecorator:
    decorates: App\DI\Bar\BarInterface
```
I've got
```
Circular reference detected for service "App\DI\Foo\FooDecorator", path: "App\DI\Foo\FooDecorator -> App\DI\Foo\FooDecorator".
```
If I comment `decorates: App\DI\Bar\BarInterface` line - all works fine
Explicit setting 
```
arguments:
    $decorated: '@App\DI\Bar\BarDecorator.inner'
```
also works

**How to reproduce**  
https://github.com/BoShurik/symfony-di-issue

**Comment by nicolas-grekas** [comment]
I think this has been fixed by #28060
I cannot reproduce it when I upgrade to 4.1.4@dev

**Comment by BoShurik** [comment]
Yes, it works. Thanks!
